crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -405,6 +405,7 @@ pub async fn execute_checkpoint(
     );
     metrics.checkpoint_transaction_count.report(tx_count as u64);
 
+    let end_of_epoch = checkpoint.summary().end_of_epoch_data.is_some();
```

Comment:
You don't have to pass this in. It can be checked inside execute_transactions

Reply:
Checking it inside execute_transactions means we have to iterate through all the transactions checking for an end of epoch tx (or assuming it's always the last tx), which I was trying to prevent. But I suppose it's simpler that way. Will fix